pkg/cmd/adm/unregister_member_test.go

```diff
@@ -36,9 +39,53 @@ func TestUnregisterMemberWhenAnswerIsY(t *testing.T) {
 	assert.Contains(t, term.Output(), "Delete Member cluster stated above from the Host cluster?")
 	assert.Contains(t, term.Output(), "The deletion of the Toolchain member cluster from the Host cluster has been triggered")
 	assert.NotContains(t, term.Output(), "cool-token")
-
 	AssertDeploymentHasReplicas(t, fakeClient, hostDeploymentName, 1)
-	assert.Equal(t, 2, numberOfUpdateCalls)
+}
+
+func TestUnregisterMemberWhenRestartError(t *testing.T) {
+	// given
+	toolchainCluster := NewToolchainCluster(ToolchainClusterName("member-cool-server.com"))
+	hostDeploymentName := test.NamespacedName("toolchain-host-operator", "host-operator-controller-manager")
+
+	newClient, fakeClient := NewFakeClients(t, toolchainCluster)
+	numberOfUpdateCalls := 0
+	fakeClient.MockUpdate = whenDeploymentThenUpdated(t, fakeClient, hostDeploymentName, 1, &numberOfUpdateCalls)
```

Comment:
please, keep an eye on what I mentioned/suggested in the comment https://github.com/kubesaw/ksctl/pull/79#discussion_r1850376661
the `hostDeploymentName` and the mock implementation are not really needed
```suggestion
	newClient, fakeClient := NewFakeClients(t, toolchainCluster)
```
and apply the same also to other test cases

Reply:
updated